mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebChromeClient.java

```diff
@@ -55,6 +70,106 @@ public boolean onConsoleMessage(final ConsoleMessage message) {
         return super.onConsoleMessage(message);
     }
 
+    public void openFileChooser(ValueCallback<Uri> uploadFile, String acceptType, String capture) {
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+
+        Log.d("Opening file chooser");
+
+        try {
+            mIitc.startActivityForResult(intent, new FileRequestHandlerBelowLollipop(uploadFile));
+        } catch (final ActivityNotFoundException e) {
+            Toast.makeText(mIitc, "No activity to select a file found." +
+                    "Please install a file browser of your choice!", Toast.LENGTH_LONG).show();
+        }
+    }
+
+    private class FileRequestHandlerBelowLollipop implements IITC_Mobile.ResponseHandler {
+        FileRequestHandlerBelowLollipop(ValueCallback<Uri> filePathCallback) {
+            this.filePathCallback = filePathCallback;
+        }
+
+        ValueCallback<Uri> filePathCallback;
+
+        @Override
+        public void onActivityResult(final int resultCode, final Intent data) {
+            mIitc.deleteResponseHandler(this); // to enable garbage collection
+
+            Uri uri = data.getData();
+            if (uri != null) {
+                this.filePathCallback.onReceiveValue(uri);
+            }
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    private String[] convertToMimeTypes(String[] acceptTypes) {
+        Set<String> results = new HashSet<>();
+        MimeTypeMap mtm = MimeTypeMap.getSingleton();
+        for (String acceptType : acceptTypes) {
+            String convertedType = "*/*";
+            if (acceptType != null) {
+                if (acceptType.contains("/") && mtm.hasMimeType(acceptType)) {
```

Comment:
All the code looks fine to me.
But still do not work.
I suspect that issue can be in `hasMimeType`.
How we could to debug this?

Reply:
Well, I've managed to debut it.
And it's right - `mtm.hasMimeType("application/json")` is `false`.